### ISSUE TITLE
cypher file option?!

### ISSUE THREAD

**Issue by davidhbigelow** [jexp/neo4j-shell-tools#46]
It does not appear to be possible to reference a cypher file for the command line instead of the cypher statement directly.

Also - the cypher statement appears to only work if on the SAME LINE as the command line...  So coping and pasting from a human readable form to command line -- or a shell file -- hurts (a lot).

**Comment by jexp** [comment]
Both great ideas

I want to extend these tools massively anyway

Michael

Von meinem iPhone gesendet

**Comment by davidhbigelow** [comment]
I am working on a _very_ simple generator using Tcl -- Yea I know... :(  -
testing similar approach for a more scriptable importer.  (bouncing ideas
around with Dave Fauth a bit).

BUT works pretty well so far...  Still playing around...

[image: Inline image 1]

On Sat, Oct 18, 2014 at 8:35 PM, Michael Hunger notifications@github.com
wrote:

> Both great ideas
> 
> I want to extend these tools massively anyway
> 
> Michael
> 
> Von meinem iPhone gesendet
> 
> > Am 18.10.2014 um 08:11 schrieb David Bigelow notifications@github.com:
> > 
> > It does not appear to be possible to reference a cypher file for the
> > command line instead of the cypher statement directly.
> > 
> > Also - the cypher statement appears to only work if on the SAME LINE as
> > the command line... So coping and pasting from a human readable form to
> > command line -- or a shell file -- hurts (a lot).
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jexp/neo4j-shell-tools/issues/46#issuecomment-59635107
> .

## 

David Bigelow
Simplified Logic, Inc.
C: 317-431-5454

**Comment by sergeiwallace** [comment]
Has the feature allowing cypher query files been implemented yet? I have a ton of cypher query files that I'd like to run (and export results to csv) with a (Windows) batch script.

**Comment by jexp** [comment]
In shell you should be able to just use `neo4j-shell --file import.cypher`

**Comment by sergeiwallace** [comment]
Can you use your cypher-import tool with cql/cypher query files though? For example, would the following command work? 

`Neo4jShell> import-cypher -o out.csv query.cql`

where the query.cql file queries an already loaded database (i.e. not using your -i flag) and the returned rows are piped to the out.csv file.

**Comment by jexp** [comment]
no, you would have a import.txt file that contains the commands

```
import-cypher -o out1.csv MATCH ....
import-cypher -o out1.csv MATCH ....
```

If you have  the same query repeatedly executed with different parameters you can use `-i in.csv`
